src/doom/g_game.c

```diff
@@ -409,27 +423,48 @@ void G_BuildTiccmd (ticcmd_t* cmd, int maketic)
 	forward -= forwardmove[speed]; 
     }
 
-    if (joyymove < 0) 
-        forward += forwardmove[speed]; 
-    if (joyymove > 0) 
-        forward -= forwardmove[speed]; 
+    if (use_analog && joyymove)
+    {
+        joyymove = joyymove > FRACUNIT ? FRACUNIT : joyymove;
```

Comment:
Add parenthesis around comparison so the reader doesn't have to think about operator precedence.